luarules/gadgets/unit_inherit_creation_xp.lua

```diff
@@ -82,70 +79,69 @@ local lastRunFrame = 0
+				if parentsInheritXP[parentDefID] ~= nil and string.find(parentsInheritXP[parentDefID], "DRONE") then
+					childrenWithParents[unitID] = {
+						unitid = unitID,
+						parentunitid = parentID,
+						parentxpmultiplier = calculatePowerDiffXP(unitID, parentID),
+						childinheritsXP = childrenInheritXP[unitDefID],
+						childtype = "DRONE",
+					}
 				end
-				if childrenWithParents[unitID] then
-					parentID = childrenWithParents[unitID].parentunitid
-					local oldXP = oldChildXPValues[unitID] or 0
-					local newXP = Spring.GetUnitExperience(unitID) or 0
-					local parentXP = Spring.GetUnitExperience(parentID) or 0
-					local multiplier = childrenWithParents[unitID].parentxpmultiplier
-					local gainedXP = (parentXP*10000000 + ((newXP-oldXP)*10000000*multiplier))/10000000 --10000000 is to prevent very small numbers being lost during calculation
-					oldChildXPValues[unitID] = newXP
-					Spring.SetUnitExperience(parentID, gainedXP)
+			end
+			if Spring.GetUnitRulesParam(unitID, "parent_unit_id") then --estabalishes unit_explosion_spawner parenthood
+				parentID = Spring.GetUnitRulesParam(unitID, "parent_unit_id")
+				parentDefID = Spring.GetUnitDefID(parentID)
+				if parentsInheritXP[parentDefID] ~= nil and string.find(parentsInheritXP[parentDefID], "BOTCANNON") then
+					childrenWithParents[unitID] = {
+						unitid = unitID,
+						parentunitid = parentID,
+						parentxpmultiplier = calculatePowerDiffXP(unitID, parentID),
+						childinheritsXP = childrenInheritXP[unitDefID],
+						childtype = "BOTCANNON",
+					}
+				end
+			end
+			if childrenWithParents[unitID] then
+				parentID = childrenWithParents[unitID].parentunitid --sets parentID if it's not already set
+				parentDefID = Spring.GetUnitDefID(parentID) -- gets the parentDefID
+			end
+			if parentID ~= nil and childrenInheritXP[parentDefID] and childrenWithParents[unitID] then --if the parent has the unitdef, set childxp to parent xp.
+				local parentTypes = childrenInheritXP[parentDefID]
+				if string.find(parentTypes, childrenWithParents[unitID].childtype) then -- if child is correcty type, set xp
+					local parentXP = Spring.GetUnitExperience(parentID)
+					Spring.SetUnitExperience(unitID, parentXP)
+					oldChildXPValues[unitID] = parentXP	--add parent xp to the oldxp value to exclude it from inheritance
 				end
 			end
+			if inheritChildrenXP[unitDefID] then -- if parent inherits xp, then add to list of units to inherit xp
+				xpGainParents[unitID] = true
+			end
+			if not xpGainParents[unitID] and not childrenWithParents[unitID] then --if not a parent or a child of a parent who inherits xp, add to ignore list
+				ignoreList[unitID] = true
+			end
+			initializeList[unitID] = nil -- this concludes innitialization
+		end
+		
+		
+		for unitID, value in pairs(childrenWithParents) do
+			parentID = childrenWithParents[unitID].parentunitid
+			local oldXP = oldChildXPValues[unitID] or 0
+			local newXP = Spring.GetUnitExperience(unitID) or 0
```

Comment:
Functions called within a tight loop multiple times are worth localizing

Reply:
Just asking this for learning purposes: 

Is it because it keeps the stuff in memory closer together to make them faster to access thus making it more performant?